.changeset/bright-olives-wonder.md

```diff
@@ -0,0 +1,7 @@
+---
+"wrangler": patch
+---
+
+fix: Stop unnecessarily amalgamating duplicate headers in Pages Functions
+
+Previously, `set-cookie` mulitple headers would be combined because of unexpected behavior in [the spec](https://github.com/whatwg/fetch/pull/1346).
```

Comment:
```suggestion
Previously, `set-cookie` multiple headers would be combined because of unexpected behavior in [the spec](https://github.com/whatwg/fetch/pull/1346).
```